lib/src/modules/conditional.dart

```diff
@@ -890,6 +889,43 @@ enum ConditionalType {
   priority
 }
 
+/// Shorthand for a [Conditional] inside a [Case] block.
+/// Returns a [Logic] signal based on the expression and conditions.
+/// The width of the result is determined by the width of the value of the
+/// conditions.
+Logic cases(Logic expression, Map<Logic, dynamic> conditions, {int? width}) {
+  for (final condition in conditions.entries) {
+    int? inferredWidth;
+    if (condition.value is Logic) {
+      inferredWidth = (condition.value as Logic).width;
+    } else if (condition.value is LogicValue) {
+      inferredWidth = (condition.value as LogicValue).width;
+    }
+
+    width ??= inferredWidth;
```

Comment:
Feels like something's not right here.  If the `inferredWidth` doesn't match a provided `width` that is not null, we should throw an exception.  After this line, we're overriding the value in `width`.